tests/conftest.py

```diff
@@ -29,6 +49,84 @@ def sapinvoices_client(sapinvoices_app):
     return sapinvoices_app.test_client()
 
 
+@pytest.fixture
+def ecs_client(mock_ecs_cluster, mock_ecs_task_definition, mock_ecs_network_config):
+    return ECSClient()
+
+
+@pytest.fixture
+def mock_boto3_ecs_client():
+    with patch("webapp.utils.aws.ecs.boto3.client") as mock_client:
+        yield mock_client.return_value
```

Comment:
Truthfully, I just took this pattern from a fixture I saw in [geo-harvester](https://github.com/MITLibraries/geo-harvester/blob/main/tests/conftest.py#L140-L143). 

To test this theory, I did the following: 
1. Create `webapp/utils/aws/also_ecs.py`.
2. Create an `AlsoECSClient` class:
   ```python
   @define
   class AlsoECSClient:
       cluster: str = field(factory=lambda: Config().ALMA_SAP_INVOICES_ECS_CLUSTER)
       task_definition: str = field(
        factory=lambda: Config().ALMA_SAP_INVOICES_ECS_TASK_DEFINITION
       )
       network_configuration: dict = field(
           factory=lambda: Config().ALMA_SAP_INVOICES_ECS_NETWORK_CONFIG
       )
       container: str = field(factory=lambda: Config().ALMA_SAP_INVOICES_ECR_IMAGE_NAME)

       @property
       def client(self) -> "ECSClientType":
           return boto3.client("ecs")

       def run(self):
           return self.client.run_task(
               cluster=self.cluster,
               launchType="FARGATE",
               networkConfiguration=self.network_configuration,
               overrides={},
               taskDefinition=self.task_definition,
           )
      ```
3. Create a simple test: 
   ```python
   def test_also_ecs_client(mock_boto3_ecs_client):
       also_ecs_client = AlsoECSClient()
       also_ecs_client.run()
   ```

The test runs without error, indicating that the `boto3` client assigned to `AlsoECSClient.client` was still patched.

---

I made the following changes:
1. Renamed fixture `mock_boto3_ecs_client` -> `mock_boto3_client` to avoid confusion on what this fixture does (i.e., patches any `boto3.client` commands--not specific to any service). 
2. Renamed fixture `ecs_client_run_task_response_success` ->  `boto3_ecs_client_run_task_response_success` to clarify that the fixture emulates the response from `boto3.client.ECS.run_task` (not `webapp.utils.aws.ECSClient`). 

Could look into using stubbers instead but for the time being, if these changes are sufficient, I'd like to table making further updates on this. Let me know what you think!

Reply:
Oh embarassing!  My memory is worthless.  Looks familiar... now.  

Re #1: good call (and probably could update in GeoHarvester).  It's definitely mocking the client, which is kind of blank until initialized with a particular client type.

Re #2: Looks good!

Yep, I'm 100% good with tests as-is.